scripts/fuse/README.MD

```diff
@@ -0,0 +1,166 @@
+|       |   |-- @elektra.value
+|       |   `-- leaf
+|       |-- info
+|       `-- person
+|           `-- name
+
+```
+as obtained with
+```sh
+tree -L 4 mountpoint
+```
+
+
+
+## Assumptions and special behaviour
+
+Elektra does not map perfectly to the structure of a classical filesystem.
+Whenever deviations are necessary, the goal is to maximize usability and adhere to the expectations of common unix tools.
+
+### Intermediate directories
+
+In Elektra, there is no need to have a `path` of continous keys between ancestors.
+E.g. the existence of `user:/a` and `user:/a/b/c` does not imply the existence of `user:/a/b`.
+Such intermediate keys are mapped to "virtual" directories, s.t. navigation in an interactive manner is possible and the directory tree is continous.
+
+A side effect is that creating/deleting a single key can create/delete multiple "virtual" directories at once.
+Another is that copies created with `cp -r` will not have these vacuous intermediaries but instead empty keys.
+
+### Keys that act both as file and directory
+
+In Elektra, there are only keys and no distinction between directories and files are made. To enable a bidirectional mapping into a classical file system, the following mechanism is employed:
+
+If a key (`A`) exists, and has some child key (`B`), `A` is treated as a directory containing `B` and a special file `@elektra.value`. This file is used to access the value of `A`.
+
+Some side effects arise:
+E.g. recursively deleting directories with `rm -r` containing `@elektra.value` files raises (task uncritical) errors when attempting to delete a parent directory which was already deleted with the removal of an `@elektra.value` file directly below.
+
+### "Leaf"-Keys that act as as directories
+
+Keys at the bottom of the hierarchy are normally treaded as files. However in certain cases, like explicitly creating directories using `mkdir`, this would result in files beeing displayed, making it impossible to expand the hierarchy using standard tools.
+Therefore, when a directory is created, the special meta-key `fuse-directory` is set to override this behaviour.
+
+### Filesystem attributes
+
+The filesystem attributes are sourced from the file obtained by `kdb file`.
+For some keys, no such file is returned, or the returned file does not exist in actuality. In these cases, fake attributes are used to preserve usability.
+
+To increase compatibility, currently the timestamps from these files are not used, as this prompts software like `vim` to produce warnings/errors.
+
+## Meta-Keys
+
+Meta-keys are mapped bidirectionally to extended file attributes (xattrs).
+To enable interoperability with other tools, the `meta:/` prefix is optional when reading/writing accessing meta keys. The listing of attributes however will show them prefix-less.
+
+Furthermore, no binary values for these attributes are allowed. (As libelektra does not accept those).
+
+See `xattr(1)` on how to read/write/list these attributes.
```

Comment:
The man page is not present in Docker and on my system (7) opens by default which is full of descriptions of file-system-specific behavior. A small example how to get+set will not harm, e.g.:

```
xattr -w abc def person/name
kdb meta-get system:/person/name abc
```